### PR TITLE
Add Reference Count to Profile Collection

### PR DESCRIPTION
**Pull request by waahm7**
*Description of changes:*
- Adds reference counting to Profile collection

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/aws_profile.c [hunk 1]
```diff
@@ -592,12 +594,11 @@ static void s_profile_hash_table_value_destroy(void *value) {
  */
 
 void aws_profile_collection_destroy(struct aws_profile_collection *profile_collection) {
-    if (profile_collection == NULL) {
-        return;
-    }
+    aws_profile_collection_release(profile_collection);
+}
 
+void aws_profile_collection_destroy_internal(struct aws_profile_collection *profile_collection) {
```
[the remark this answers is not in the document]
Comment (waahm7) [reply source/aws_profile.c]: Thanks, fixed.

File: source/aws_profile.c [hunk 2]
```diff
@@ -781,7 +784,7 @@ struct aws_profile_collection *aws_profile_collection_new_from_merge(
     return merged;
 
 cleanup:
-    aws_profile_collection_destroy(merged);
+    aws_profile_collection_release(merged);
```
[the remark this answers is not in the document]
Comment (waahm7) [reply source/aws_profile.c]: Thanks, fixed.

File: source/aws_profile.c [hunk 3]
```diff
@@ -1238,11 +1246,25 @@ static struct aws_profile_collection *s_aws_profile_collection_new_internal(
...
 
+struct aws_profile_collection *aws_profile_collection_acquire(struct aws_profile_collection *collection) {
+    if (collection != NULL) {
+        aws_ref_count_acquire(&collection->ref_count);
+    }
+
+    return collection;
+}
+
+void aws_profile_collection_release(struct aws_profile_collection *collection) {
```
[the remark this answers is not in the document]
Comment (waahm7) [reply source/aws_profile.c]: Thanks, fixed.

File: source/aws_profile.c [hunk 4]
```diff
@@ -592,12 +594,11 @@ static void s_profile_hash_table_value_destroy(void *value) {
  */
 
 void aws_profile_collection_destroy(struct aws_profile_collection *profile_collection) {
-    if (profile_collection == NULL) {
-        return;
-    }
+    aws_profile_collection_release(profile_collection);
+}
 
+static void aws_profile_collection_destroy_internal(struct aws_profile_collection *profile_collection) {
```
Comment (TingDaoK) [inline source/aws_profile.c:600]: trivial: we named the static functions to start with `s_*`
```suggestion
static void s_profile_collection_destroy_internal(struct aws_profile_collection *profile_collection) {
```

**Review by TingDaoK — APPROVED** [review]
fix and ship

File: source/aws_profile.c (hunk 4 above)
Comment (waahm7) [reply source/aws_profile.c]: Sorry, I have updated it.

**Pull request merged**